pages/index.vue

```diff
@@ -147,25 +179,69 @@ export default defineComponent({
      * @todo 1. Send the request to update the task to the backend server.
      * @todo 2. Update the task in the dom.
      */
-    updateTask(_index, _id) {},
+    async updateTask(_index, _id) {         
+      console.log("updatingg")
+      console.log("data67");
+      const data={title: this.editTitle,}
+      const headers={Authorization: 'Token ' + this.$store.getters.token}
+
+      if(this.todos[_index].title!= ''){
+       await this.$axios.patch(`https://todo-app-csoc.herokuapp.com/todo/${_id}/`,data,{headers})
+        .then((response) => 
+        {
+        console.log("data66");
+        console.log(response);
+        console.log(this.todos[_index].editing)
+        console.log("data77");
+        this.todos[_index].title=this.editTitle
+        this.editTitle=''
+        this.$toast.success('Task Updated')
+        this.todos[_index].editing = !this.todos[_index].editing;
+        })
+          .catch((err) => {
+          this.$toast.error("Error!..")
+        })
+      }
+        this.todos[_index].editing = !this.todos[_index].editing;
```

Comment:
Why are you adding this here? This is already present on line 199. This might be the reason you are facing difficulties while updating todos.